bagua/torch_api/distributed.py

```diff
@@ -180,6 +184,10 @@ def with_bagua(  # pytype: disable=module-attr
             ...    )
         """
 
+        self.name = "{}_{}".format(
```

Comment:
it seems to me this needs to be in `with_bagua`, not in `bagua_build_params`?

The `bagua_build_params` could be called multiple times in an `Algorithm`

Reply:
This sentence is indeed in the with_bagua code block, the code needs to be expanded

Reply:
got it

Reply:
don't use self.name,  use something like self._bagua_module_name, to avoid name clash

Reply:
OK